### ISSUE TITLE
`didReceiveAttrs` requires `this._super` call.

### ISSUE THREAD

**Issue by mmelvin0** [emberjs/ember.js#11992]
When didReceiveAttrs fires on first render, this.get('value') === this.getAttr('value').
When didReceiveAttrs fires on re-renders, this.get('value') === (previous value) and this.getAttr('value') === (current value).

**Comment by rwjblue** [comment]
* For which `Ember.VERSION`?
* Can you provide a JSBin / ember-twiddle reproduction?

**Comment by mmelvin0** [comment]
1.13.5 and 1.13.6
http://ember-twiddle.com/7642c2bfe8a8868a3b5f

**Comment by pixelhandler** [comment]
This page seems to be the only place that `didReceiveAttrs` is documented http://emberjs.com/blog/2015/06/12/ember-1-13-0-released.html#toc_component-lifecycle-hooks

I don't see `getAttr` listed in the API docs at http://emberjs.com/api/classes/Ember.Component.html

Does seem like a bug for `this.get('value')` though. However since `getAttr` and `didReceiveAttrs` don't seem to be documented in the api docs for 1.13.x I'm confused as to what I should expect. Seems that both the guides and api docs lack explanation of the lifecycle hooks. I'm curious to how these landed in the release without any docs (aside from a blog post announcing them). 

Maybe we're only supposed to use `getAttr` in `didRecieveAttrs` I see that the blog post mentions `this.get` can be expected to be accurate after `didInitAttrs` So in my opinion this does seem to be a BUG.

**Comment by rwjblue** [comment]
@pixelhandler:

> Seems that both the guides and api docs lack explanation of the lifecycle hooks. I'm curious to how these landed in the release without any docs (aside from a blog post announcing them).

* We did miss adding some docs for features as they were added during the 1.13/glimmer work, but we are working on updated guides and API docs for 2.0/2.1 that will correct that mistake.  We have also discussed and decided that we will not be landing new features/API's without docs in the future (this was not a rule that existed during any of the 1.x series but we feel that it is very important for 2.x+).
* Usage of `getAttr`/`setAttr` and the various hooks isn't really "promoted" yet (but it is definitely important that they work), because significant portions of functionality are still not landed (likely some time in 2.1 and 2.2).

---

This is indeed a bug, `didReceiveAttrs` is firing (and the new attributes are set in `this.attrs`) prior to the values being proxied back to the root of the component.  Most likely this is an ordering problem, but I am not 100% sure just yet...

**Comment by rwjblue** [comment]
I believe this is fixed in 1.13.8. Can you double check (I'm mobile ATM)?

**Comment by rwjblue** [comment]
Just confirmed that this works properly if you call `_super` in `didReceiveAttrs`: http://ember-twiddle.com/e961367026f0885e6ce5.

While I absolutely suggest that whenever you implement a framework hook you should call `_super`, it should not be a requirement.

**Comment by rwjblue** [comment]
I'm updating the title of this issue to track that `_super` requirement...

**Comment by shaunc** [comment]
Ah -- very good. @mmun had me take those out of my original pull request for ember-collection. Just to confirm: always call `_super` in all hooks? (Great if that could be said firmly in the docs, if true.)

**Comment by mmun** [comment]
tl;dr Robert is right. Adding `_super` is fine. It's nice to avoid _super because it has a small performance implication (it creates a weird _superWrapper thing) and the hooks are _supposed_ to have the contract that they are not necessary to be super'd.

**Comment by shaunc** [comment]
Ok ... I understand: as the new title suggests -- this is still a bug, but one that can be more elegantly avoided that via an ugly hack. :).
